HPXMLtoOpenStudio/resources/hpxml_schematron/EPvalidator.xml

```diff
@@ -1314,6 +1333,34 @@
     </sch:rule>
   </sch:pattern>
 
+  <sch:pattern>
+    <sch:title>[GeothermalLoop]</sch:title>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Systems/h:HVAC/h:HVACPlant/h:GeothermalLoop'>
+      <sch:assert role='ERROR' test='count(h:LoopConfiguration) = 1'>Expected 1 element(s) for xpath: LoopConfiguration</sch:assert>
+      <sch:assert role='ERROR' test='h:LoopConfiguration[text()="vertical"] or not(h:LoopConfiguration)'>Expected LoopConfiguration to be 'vertical'</sch:assert>
+      <sch:assert role='ERROR' test='count(h:LoopFlow) &lt;= 1'>Expected 0 or 1 element(s) for xpath: LoopFlow</sch:assert>
+      <sch:assert role='ERROR' test='number(h:LoopFlow) &gt; 0 or not(h:LoopFlow)'>Expected LoopFlow to be greater than 0</sch:assert>
+      <sch:assert role='ERROR' test='count(h:BoreholesOrTrenches/h:Count) &lt;= 1'>Expected 0 or 1 element(s) for xpath: BoreholesOrTrenches/Count</sch:assert>
+      <sch:assert role='ERROR' test='number(h:BoreholesOrTrenches/h:Count) &gt;= 1 or not(h:BoreholesOrTrenches/h:Count)'>Expected BoreholesOrTrenches/Count to be greater than or equal to 1</sch:assert>
+      <sch:assert role='ERROR' test='number(h:BoreholesOrTrenches/h:Count) &lt;= 10 or not(h:BoreholesOrTrenches/h:Count)'>Expected BoreholesOrTrenches/Count to be less than or equal to 10</sch:assert>
+      <sch:assert role='ERROR' test='count(h:BoreholesOrTrenches/h:Length) &lt;= 1'>Expected 0 or 1 element(s) for xpath: BoreholesOrTrenches/Length</sch:assert>      
```

Comment:
Do we allow LoopFlow, Count, and Length to be independently optional? E.g., if someone tells us Count (or Length) only, can we default the other two? Or if they provide two of the three, can we default the other one?

I think we'd like to, since some applications may not collect all three inputs, but I can't tell if the code will handle it. Maybe we need some additional tests in test_defaults.rb.

Reply:
I guess I'm confused here. Doesn't this test in `test_defaults.rb` already show that they can be independently optionally? Or are you suggesting testing every combination?

```
# Test defaults
hpxml_bldg.geothermal_loops[0].loop_flow = nil
hpxml_bldg.geothermal_loops[0].num_bore_holes = nil
hpxml_bldg.geothermal_loops[0].bore_spacing = nil
hpxml_bldg.geothermal_loops[0].bore_length = nil
...
```

Reply:
No, that is testing all the inputs being defaulted together, not being defaulted independently.